dashboard/modules/usage_stats/usage_stats_head.py

```diff
@@ -104,8 +104,13 @@ async def run(self, server):
             return
         else:
             logger.info("Usage reporting is enabled.")
+            # Wait for 2 minutes to send the first report
```

Comment:
Alternatively, we can keep polling until the information exists, but I think a 2 minute delay is good enough. It also potentially allow us to collect library usage in the first report. Also users has the chance to stop the cluster if they don't want usage collection.

Reply:
This raises an interesting sampling question. Many clusters are short lived. If we only report at 2 min, we might miss short lived clusters. On the other hand, this gives some time for libraries to get imported, etc., so maybe some delay is a must.

Btw why 2min and not say 30s?

Reply:
Yea, we were thinking clusters < 2 min might not be interesting to us but that might be wrong.

2 min is chosen because its long enough so we can collect interesting thing to report in the first report but short enough that we won't miss many clusters. But that doesn't mean we cannot change to 1m or 30s.

There is another alternative: we send first report immediately, after x (2) min, we send the second, then after a random delay, we send the third. This will increase QPS a bit but might be OK.

Reply:
I think this is ok, but I'd prefer something like 1 min which seems easier to explain than 2 min.